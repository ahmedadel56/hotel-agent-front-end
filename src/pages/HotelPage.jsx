import React from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import HotelList from '../components/HotelList';

const HomePage = () => {
  const { data, status } = useSelector((state) => state.Hotel);
  return (
    <div className="featured-container">
      <div className="list-heading">
        <h1>Hotel List</h1>
      </div>
      {status === 'successfully loaded' && (
        <>
          <Link to="/add-hotel">
            <button type="button" className="addNewBtn ">
              Add new Hotel
            </button>
          </Link>
          <div className="hotel-list">
            {data.map((hotel) => (
              <HotelList hotel={hotel} key={hotel.id} />
            ))}
          </div>
        </>
      )}
    </div>
  );
};

export default HomePage;
