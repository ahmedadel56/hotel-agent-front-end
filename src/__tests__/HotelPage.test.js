import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import HotelPage from '../pages/HotelPage';

it('Check if Hotel page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <HotelPage />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});
