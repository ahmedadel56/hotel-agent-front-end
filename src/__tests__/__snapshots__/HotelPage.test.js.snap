// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Hotel page renders correctly 1`] = `
<div
  className="featured-container"
>
  <div
    className="list-heading"
  >
    <h1>
      Hotel List
    </h1>
  </div>
</div>
`;
